### PR TITLE
Include a nil check for the snapshotContent

### PR DESCRIPTION
**Pull request by huffmanca**
/kind bug

**What this PR does / why we need it**:
It's currently possible for the snapshotController to crash when checking snapshot Finalizers if the VolumeSnapshotContent is nil. This PR introduces a simple nil check to prevent the panic.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/common-controller/snapshot_controller.go [hunk 1]
```diff
@@ -266,8 +266,10 @@ func (ctrl *csiSnapshotCommonController) checkandRemoveSnapshotFinalizersAndChec
 		}
 
 		if !inUse {
-			klog.V(5).Infof("checkandRemoveSnapshotFinalizersAndCheckandDeleteContent: Set VolumeSnapshotBeingDeleted annotation on the content [%s]", content.Name)
-			ctrl.setAnnVolumeSnapshotBeingDeleted(content)
+			if content != nil {
+				klog.V(5).Infof("checkandRemoveSnapshotFinalizersAndCheckandDeleteContent: Set VolumeSnapshotBeingDeleted annotation on the content [%s]", content.Name)
+				ctrl.setAnnVolumeSnapshotBeingDeleted(content)
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_controller.go]: This is called in multiple places.  Can you add a nil check inside setAnnVolumeSnapshotBeingDeleted()?
I actually just ran into this yesterday so I added it here:
https://github.com/kubernetes-csi/external-snapshotter/pull/261/files#diff-214a2fc68717c170415a3578536d3b74R1281

I'll remove my change once your PR is merged.

Comment (huffmanca) [reply pkg/common-controller/snapshot_controller.go]: I've updated this to also include the check inside of `setAnnVolumeSnapshotBeingDeleted()`. I was hitting the panic on line 270 above, when we were printing out the debug statement.

I can remove the name of the content, since it should print this out within `setAnnVolumeSnapshotBeingDeleted` immediately after it checks for nil now.

Comment (xing-yang) [reply pkg/common-controller/snapshot_controller.go]: This nil check on line 269 can be removed now since you moved the check inside setAnnVolumeSnapshotBeingDeleted()

Comment (huffmanca) [reply pkg/common-controller/snapshot_controller.go]: I removed the nil check here, but I also updated the debug logging to not print the content name. That's where I was hitting the panic

File: pkg/common-controller/snapshot_controller.go [hunk 2]
```diff
@@ -1281,6 +1281,9 @@ func (ctrl *csiSnapshotCommonController) getSnapshotFromStore(snapshotName strin
 func (ctrl *csiSnapshotCommonController) setAnnVolumeSnapshotBeingDeleted(content *crdv1.VolumeSnapshotContent) error {
 	// Set AnnVolumeSnapshotBeingDeleted if it is not set yet
 	if !metav1.HasAnnotation(content.ObjectMeta, utils.AnnVolumeSnapshotBeingDeleted) {
+		if content == nil {
+			return nil
+		}
```
Comment (xing-yang) [inline pkg/common-controller/snapshot_controller.go]: This check should be moved before the if block.  Otherwise we'll still get a nil exception on line 1283 as it references the content.  I'm going to revert this PR.  Can you please submit a different PR?

Comment (xing-yang) [reply pkg/common-controller/snapshot_controller.go]: Fixed in https://github.com/kubernetes-csi/external-snapshotter/pull/279
